### PR TITLE
fix(ivy): Ensure proper namespace is used to create elements in JIT

### PR DESCRIPTION
**Pull request by benlesh**
Updates render to properly use the namespace URL that Ivy passes around rather than use it to look up the namespace URL ala ViewEngine. 

Resolves [FW-811](https://angular-team.atlassian.net/browse/FW-811)

### CODE REVIEW

**Comment by AndrewKushnir** [comment]
@benlesh it looks like you've created this PR on top of the master while it was in a state where "integration_test" was broken. Please rebase it from the latest master to make "integration_test" pass. Thank you.

File: packages/platform-browser/src/dom/dom_renderer.ts [hunk 1]
```diff
@@ -111,7 +111,9 @@ class DefaultDomRenderer2 implements Renderer2 {
 
   createElement(name: string, namespace?: string): any {
     if (namespace) {
-      return document.createElementNS(NAMESPACE_URIS[namespace], name);
+      // In cases where Ivy (not ViewEngine) is giving us the actual namespace, the look up by key
+      // will result in undefined, so we just return the namespace here.
+      return document.createElementNS(NAMESPACE_URIS[namespace] || namespace, name);
```
Comment (kara) [inline packages/platform-browser/src/dom/dom_renderer.ts]: There are two other places where we look the namespace up in this file - `setAttribute` and `removeAttribute`. Can you update those too?

**Review by kara — CHANGES_REQUESTED** [review]

**Comment by AndrewKushnir** [comment]
@kara @benlesh I think we may also need to check how VE and Ivy handle the case when invalid/unsupported namespace is specified and align this behavior (and have a test to capture this). My understanding is that we'll see a difference after this change due to the fact that we return namespace as is in case it's not registered in `NAMESPACE_URIS`.

**Review by mhevery — DISMISSED** [review]
Presumably there is a failing tests which this unblocks. I don't see any changes to the tests.

**Comment by trotyl** [comment]
What if some custom renderers do need the raw namespace text? Renderer is a public API, how could a breaking change being introduced directly?

File: packages/platform-browser/src/dom/dom_renderer.ts (hunk 1 above)
Comment (kara) [reply packages/platform-browser/src/dom/dom_renderer.ts]: Update based on offline discussion: 
- add TODO comments to `setAttribute` and `removeAttribute`
- remove fixme for test

File: packages/platform-browser/src/dom/dom_renderer.ts [hunk 2]
```diff
@@ -154,6 +156,8 @@ class DefaultDomRenderer2 implements Renderer2 {
   setAttribute(el: any, name: string, value: string, namespace?: string): void {
     if (namespace) {
       name = `${namespace}:${name}`;
+      // TODO(benlesh): Ivy may cause issues here because it's passing around
+      // full URIs for namespaces, therefor this lookup will fail.
```
Comment (kara) [inline packages/platform-browser/src/dom/dom_renderer.ts]: Same below:

```suggestion
      // full URIs for namespaces, therefore this lookup will fail.
```

**Review by kara — APPROVED** [review]
LGTM, aside from typo

**Pull request closed without merging**